core/tests/puppeteer-acceptance-tests/user-utilities/question-admin-utils.js

```diff
@@ -0,0 +1,163 @@
+   'button#add-contribution-rights-submit-button';
+
+module.exports = class QuestionAdmin extends baseUser {
+  /**
+   * Function for navigating to the contributor dashboard admin page.
+   */
+  async navigateToContributorDashboardAdminPage() {
+    await this.goto(contributorDashboardAdminUrl);
+  }
+
+  /**
+   * Function for adding a right of reviewing questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async addReviewQuestionRights(username) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(addContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function for adding a right of submitting questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async addSubmitQuestionRights(username) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, submitQuestionRightValue);
+    await this.clickOn(addContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to display contribution rights by user.
+   * @param {string} username - the username of the user to view.
+   */
+  async viewContributionRightsForUser(username) {
+    await this.select(viewContributorFilterMethodSelect, usernameMethodValue);
+    await this.type(viewContributerUsernameInput, username);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to display reviewing questions rights by role.
+   */
+  async viewContributorReviewQuestionRightsByRole() {
+    await this.select(viewContributorFilterMethodSelect, roleMethodValue);
+    await this.select(viewContributorCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to check if the username is displayed as one of the users
+   * with rights of reviewing question.
+   * @param {string} username - the username of the user to check.
+   */
+  async expectDisplayedUsersToContainReviewQuestionRights(username) {
+    await this.page.waitForSelector(viewContributorReviewQuestionsResult);
+    const displayedUsername = await this.page.$eval(
+      viewContributorReviewQuestionsResult,
+      element => element.innerText);
+    if (displayedUsername.includes('Not-allowed')) {
+      throw new Error(
+        `${username} does not have rights for reviewing questions!`);
+    } else {
+      showMessage(
+        `${username} has rights for reviewing questions!`);
+    }
+  }
+
+  /**
+   * Function to check if the user is displayed as a question reviewer.
+   * @param {string} username - the user expected to be displayed.
+   */
+  async expectUserToBeDisplayed(username) {
```

Comment:
This name is vague because it says nothing about practice questions, so it's unclear where we're expecting the user to be displayed. See my comment in the top-level test for alternative naming.

Reply:
Done